lib/msal-core/src/authority/AuthorityFactory.ts

```diff
@@ -37,15 +22,13 @@ export class AuthorityFactory {
         if (StringUtils.isEmpty(authorityUrl)) {
             return null;
         }
-        const type = AuthorityFactory.DetectAuthorityFromUrl(authorityUrl);
+
         // Depending on above detection, create the right type.
-        switch (type) {
-            case AuthorityType.B2C:
-                return new B2cAuthority(authorityUrl, validateAuthority);
-            case AuthorityType.Aad:
-                return new AadAuthority(authorityUrl, validateAuthority);
-            default:
-                throw ClientConfigurationErrorMessage.invalidAuthorityType;
+        if (Object.keys(B2CTrustedHostList).length) {
+            return new B2cAuthority(authorityUrl, validateAuthority);
```

Comment:
Can we make this ternary check?

Reply:
Good call, will do

Reply:
Also may be consider ADFS in future and makes changes accordingly. ADFS should be a different type of authority.

Reply:
You can default it to a generic case and not B2C if you decide on instantiating all three authorities as different instances.

Reply:
I think ultimately it would make sense to collapse AAD and B2C classes into a single generic class as there is very little difference between them. Not sure how ADFS would differ at that point. But that's probably a longer topic of discussion

Reply:
Yes. If we decide no `type`: All collapse to one class; otherwise they stay separate. Deciding it now and making the changes will make it easy?